samples/bot-calling-meeting/csharp/Source/CallingBotSample/Bots/CallingBot.cs

```diff
@@ -98,21 +108,69 @@ private async Task NotificationProcessor_OnNotificationReceivedAsync(Notificatio
             // https://microsoftgraph.github.io/microsoft-graph-comms-samples/docs/articles/index.html#answer-incoming-call-with-service-hosted-media
 
             graphLogger.CorrelationId = args.ScenarioId;
+            var callId = GetCallIdFromNotification(args);
+
             if (args.ResourceData is Call call)
             {
+
                 if (args.ChangeType == ChangeType.Created && call.State == CallState.Incoming)
                 {
-                    await callService.Answer(GetCallIdFromNotification(args), audioRecordingConstants.Speech, audioRecordingConstants.PleaseRecordYourMessage);
+                    await callService.Answer(callId, audioRecordingConstants.Speech, audioRecordingConstants.PleaseRecordYourMessage);
                 }
                 else if (
                     args.ChangeType == ChangeType.Updated
-                    && call.State == CallState.Established
-                    // The below helps to distinguish between two similar Established notifications that are sent for incoming 1:1 meetings
```

Comment:
Can you explain how the check below handles these cases?

Reply:
Added comment